client/landing/stepper/declarative-flow/internals/hooks/use-step-navigation-with-tracking/index.ts

```diff
@@ -35,6 +37,17 @@ export const useStepNavigationWithTracking = ( {
 		};
 	}, [] );
 
+	const stepData = useSelect(
+		( select ) => ( select( STEPPER_INTERNAL_STORE ) as StepperInternalSelect ).getStepData(),
+		[]
+	);
+
+	/**
+	 * If the previous step is defined in the store, and the current step is not the first step, we can go back.
+	 * We need to make sure we're not at the first step because `previousStep` is persisted and can be a step from another flow or another run of the current flow.
```

Comment:
Can `previousStep` (and any stale state) be cleanup automatically / expire when starting a new flow?

Reply:
Good idea. I wouldn't worry about that now though.